common/djangoapps/enrollment/views.py

```diff
@@ -581,8 +586,19 @@ def post(self, request):
                 )
 
             enterprise_course_consent = request.data.get('enterprise_course_consent')
+            explicit_linked_enterprise = request.data.get('linked_enterprise_customer')
+            if has_api_key_permissions and enterprise_enabled() and explicit_linked_enterprise is not None:
+                kwargs = {
+                    'username': username,
+                    'course_id': unicode(course_id),
+                    'enterprise_customer_uuid': explicit_linked_enterprise,
+                }
+                consent_client = ConsentApiClient()
+                if consent_client.consent_needed(**kwargs):
+                    consent_client.provide_consent(**kwargs)
+
             # Check if the enterprise_course_enrollment is a boolean
-            if has_api_key_permissions and enterprise_enabled() and enterprise_course_consent is not None:
+            elif has_api_key_permissions and enterprise_enabled() and enterprise_course_consent is not None:
```

Comment:
small style nit, because these two if statements look really similar, it might be more readable to do:
```
if has_api_key_permissions and enterprise_enabled():
   if explicit_linked_enterprise is not None:
       // do stuff
   elif enterprise_course_consent is not None:
       // do the rest

```